### PR TITLE
Indicate which `AnnotatedTypeMirror` methods return only primary annotations

### PR DESCRIPTION
**Pull request by smillst**

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java [hunk 1]
```diff
@@ -720,12 +726,14 @@ public boolean removeAnnotationInHierarchy(AnnotationMirror a) {
    *
    * @param a an annotation from the same qualifier hierarchy
    * @return if an annotation was removed
+   * @deprecated TODO what should be called?
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java:729]: What should be called instead?

**Review by mernst — APPROVED** [review]
Thanks!

File: framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java (hunk 1 above)
Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java]: This is never called, and I can't see how it is useful, so I think it should just be removed.  I've update the Javadoc to reflect that.

**Review by mernst — DISMISSED** [review]

**Review by mernst — APPROVED** [review]

**Comment by smillst** [comment]
@mernst I think I've fixed all the Javadoc and merge problems, so this is ready for one last review.

**Review by mernst — APPROVED** [review]

**Pull request merged**